xcell/mappers/mapper_P15tSZ.py

```diff
@@ -12,25 +13,45 @@ def __init__(self, config):
          'nside':512}
         """
         self._get_Planck_defaults(config)
+        self.file_hm1 = config.get('file_hm1', self.file_map)
+        self.file_hm2 = config.get('file_hm2', self.file_map)
         self.beam_info = config.get('beam_fwhm_arcmin', 10.)
+        self.gal_mask_mode = config.get('gal_mask_mode', '0.5')
+        self.gal_mask_modes = {'0.4': 0,
+                               '0.5': 1,
+                               '0.6': 2,
+                               '0.7': 3}
 
     def _get_hm_maps(self):
         if self.hm1_map is None:
-            hm1_map = hp.read_map(self.file_map, 1)
+            hm1_map = hp.read_map(self.file_hm1, 1)
             self.hm1_map = [hp.ud_grade(hm1_map,
                             nside_out=self.nside)]
         if self.hm2_map is None:
-            hm2_map = hp.read_map(self.file_map, 2)
+            hm2_map = hp.read_map(self.file_hm2, 2)
             self.hm2_map = [hp.ud_grade(hm2_map,
                             nside_out=self.nside)]
         return self.hm1_map, self.hm2_map
 
     def get_mask(self):
         if self.mask is None:
-            mask = hp.read_map(self.file_mask)
-            mask = hp.ud_grade(mask,
-                               nside_out=self.nside)
-            self.mask = mask
+            if self.file_mask is not None:
+                self.mask = hp.read_map(self.file_mask)
+                self.mask = hp.ud_grade(self.mask,
+                                        nside_out=self.nside)
+            else:
+                self.mask = np.ones(12*self.nside**2)
+            if self.file_gp_mask is not None:
+                field = self.gal_mask_modes[self.gal_mask_mode]
+                gal_mask = hp.read_map(self.file_mask, field)
+                gal_mask = hp.ud_grade(gal_mask,
+                                       nside_out=self.nside)
+                self.mask *= gal_mask
+            if self.file_sp_mask is not None:
+                sp_mask = hp.read_map(self.file_mask, 4)
+                sp_mask = hp.ud_grade(sp_mask,
+                                      nside_out=self.nside)
+                self.mask *= sp_mask
```

Comment:
Why does the tSZ map not have different point source mask options? Remind me which point source mask you used to match the Planck paper?

Reply:
because the gal and ps masks are just different fields of the same fits file.

Reply:
oh, you're right...